### PR TITLE
zh-TW: Translate Concurrency morning

### PR DESCRIPTION
**Pull request by mgeisler**
Hi all,

This translation was prepared by our in-house localization team. Since I don't speak Chinese, please look it over and add suggestions.

Let me know how good the translation is — if it is good, I can send them bigger parts to translate!

Thanks for your help! Part of #684.

### CODE REVIEW

File: po/zh-TW.po [hunk 1]
```diff
@@ -14373,16 +14519,18 @@ msgid ""
 "* They can be used in the generic constraints as normal traits.\n"
 "  "
 msgstr ""
+"* 這些特徵可視為標記，表示該型別含有特定執行緒安全屬性。\n"
+"* 這些特徵就像一般特徵，可用於泛型條件約束。\n"
```
Comment (mgeisler) [inline po/zh-TW.po:14523]: Newline fix:

```suggestion
"* 這些特徵就像一般特徵，可用於泛型條件約束。"
```

**Comment by mgeisler** [comment]
Hey all,

I fixed the newline problem. Please read the translation use the GitHub suggestion feature to flag any problems you find. Since I don't actually read Chinese, I'll blindly commit the suggestions later (you're therefore also very much invited to committing them directly unless you feel another reviewer should read them).

Thanks for the help here!

File: po/zh-TW.po [hunk 2]
```diff
@@ -14392,39 +14540,48 @@ msgid ""
...
+msgstr "> `&T` 必須為 `Send`，`T` 才會是 `Sync`"
 
 #: src/concurrency/send-sync/sync.md:14
 msgid ""
 "This statement is essentially a shorthand way of saying that if a type is "
 "thread-safe for shared use, it is also thread-safe to pass references of it "
 "across threads."
 msgstr ""
+"這個陳述式常用來簡單表示，是否可在確保執行緒安全的情況下共用型別，同時在執行"
+"緒間傳遞型別參考。"
```
Comment (hueich) [inline po/zh-TW.po:14584]:
```suggestion
"這定義簡單的表示，若一個型別可以在確保執行緒安全的情況下被共用，這型別的參考值也可以安全的被傳遞於其他的執行緒。"
```

File: po/zh-TW.po [hunk 3]
```diff
@@ -14578,16 +14782,27 @@ msgid ""
...
 "    * `std::sync::Weak` can help."
 msgstr ""
+"* `Arc` 代表「原子參考計數」，這個 `Rc` 的執行緒安全版本會採用原子性\n"
+"運算。\n"
+"* `Arc<T>` 會實作 `Clone`，與 `T` 無關。如果 `T` 同時實作 `Send` 和 "
+"`Sync`，\n"
+"`Arc<T>` 也會實作這兩者。\n"
+"* `Arc::clone()` 會對已執行的原子性運算計費，但之後使用\n"
+" `T` 不需任何費用。\n"
+"* 留意參考循環，`Arc` 並未使用垃圾收集器進行偵測。\n"
```
Comment (hueich) [inline po/zh-TW.po:14792]:
```suggestion
"* 留意參考循環，`Arc` 並不使用垃圾收集器進行偵測。\n"
```

File: po/zh-TW.po [hunk 4]
```diff
@@ -14578,16 +14782,27 @@ msgid ""
 "detect them.\n"
 "    * `std::sync::Weak` can help."
 msgstr ""
+"* `Arc` 代表「原子參考計數」，這個 `Rc` 的執行緒安全版本會採用原子性\n"
+"運算。\n"
+"* `Arc<T>` 會實作 `Clone`，與 `T` 無關。如果 `T` 同時實作 `Send` 和 "
+"`Sync`，\n"
+"`Arc<T>` 也會實作這兩者。\n"
+"* `Arc::clone()` 會對已執行的原子性運算計費，但之後使用\n"
```
Comment (hueich) [inline po/zh-TW.po:14790]:
```suggestion
"* `Arc::clone()` 會導致執行原子性運算的費用，但之後使用得到的\n"
```

File: po/zh-TW.po [hunk 5]
```diff
@@ -14634,10 +14866,22 @@ msgid ""
 "to recover the data\n"
 "      regardless."
 msgstr ""
+"* Rust 中的 `Mutex` 就像是只有一個元素的集合，也就是受保護的資料。\n"
+"    * 必須先取得互斥鎖，才能存取受保護的資料。\n"
+"* 只要使用這個鎖，就能從 `&Mutex<T>` 取得 `&mut T`。`MutexGuard` 可確保\n"
+" `&mut T` 不會繼續留存持有的鎖。\n"
```
Comment (hueich) [inline po/zh-TW.po:14872]:
```suggestion
" `&mut T` 的壽命不會超過所持有的鎖。\n"
```

File: po/zh-TW.po [hunk 6]
```diff
@@ -14634,10 +14866,22 @@ msgid ""
...
+"* 如果 `T` 實作 `Send`，`Mutex<T>` 就會實作 `Send` 和 `Sync`。\n"
```
Comment (hueich) [inline po/zh-TW.po:14873]:
```suggestion
"* 如果 `T` 實作 `Send`，`Mutex<T>` 才會實作 `Send` 和 `Sync`。\n"
```

File: po/zh-TW.po [hunk 7]
```diff
@@ -14704,6 +14986,11 @@ msgid ""
 "* Blocks are introduced to narrow the scope of the `LockGuard` as much as "
 "possible."
 msgstr ""
+"* `v` 已同時納入 `Arc` 和 `Mutex`，因為兩者的考量互不相關。\n"
+"* 將 `Mutex` 納入 `Arc`，是在執行緒間共享可變狀態的常見模式。\n"
+"* `v: Arc<_>` 需要複製成 `v2` 克隆，才能轉移到其他執行緒。請注意，`move` 已新"
```
Comment (hueich) [inline po/zh-TW.po:14991]:
```suggestion
"* `v: Arc<_>` 需要複製成 `v2` ，才能轉移到其他執行緒。請注意，`move` 已新"
```

File: po/zh-TW.po [hunk 8]
```diff
@@ -14704,6 +14986,11 @@ msgid ""
...
+"增至 lambda 簽章。\n"
+"* 已導入阻擋功能，盡量縮小 `LockGuard` 的範圍。"
```
Comment (hueich) [inline po/zh-TW.po:14993]:
```suggestion
"* 采用區塊，盡量縮小 `LockGuard` 的範圍。"
```

**Comment by hueich** [comment]
The translations are pretty good.  But there are some domain knowledge that are missing, which sometimes leads to translating words into wrong alternatives.

File: po/zh-TW.po [hunk 9]
```diff
@@ -14301,14 +14392,37 @@ msgid ""
...
+"}\n"
+"```"
 
 #: src/concurrency/channels/bounded.md:1
 msgid "# Bounded Channels"
-msgstr ""
+msgstr "# 有界限的通道"
 
 #: src/concurrency/channels/bounded.md:3
 msgid "Bounded and synchronous channels make `send` block the current thread:"
```
[the remark this answers is not in the document]
Comment (mgeisler) [reply po/zh-TW.po:14424]: Oh, good question. I think "could" is the right word and "make" is a bit imprecise. Calling `channel.send(foo)` will block the current thread until there is space in `channel`.

**Comment by mgeisler** [comment]
> The translations are pretty good. But there are some domain knowledge that are missing, which sometimes leads to translating words into wrong alternatives.

Thanks a lot, @hueich! That's roughly what I would expect: I would not expect the translator to have any particular Rust knowledge.

Thanks for reading through it and for making fixes! I'll commit them and then you just have to let me know when you think we can merge this.

File: po/zh-TW.po [hunk 10]
```diff
@@ -14099,13 +14099,15 @@ msgstr ""
...
-msgstr ""
+msgstr "# 歡迎使用 Rust 的並行程式設計"
 
 #: src/concurrency.md:3
 msgid ""
 "Rust has full support for concurrency using OS threads with mutexes and\n"
 "channels."
 msgstr ""
+"Rust 完整支援搭配互斥鎖和通道使用 OS 執行緒\n"
+"處理並行。"
```
Comment (mgeisler) [inline po/zh-TW.po:14110]:
```suggestion
"Rust 使用 OS 執行緒搭配著互斥鎖和通道來完整支援並行處理。"
```

[the remark this answers is not in the document]
Comment (mgeisler) [reply po/zh-TW.po:14110]: Will do via a multi-line suggestion — if you drag in the left column, it's possible to make a suggestion that covers multiple lines.

File: po/zh-TW.po (hunk 9 above)
[the remark this answers is not in the document]
Comment (mgeisler) [reply po/zh-TW.po:14423]: I created #792 to improve the wording a bit.

File: po/zh-TW.po [hunk 11]
```diff
@@ -14580,8 +14579,7 @@ msgid ""
 "thread-safe for shared use, it is also thread-safe to pass references of it "
 "across threads."
 msgstr ""
-"這個陳述式常用來簡單表示，是否可在確保執行緒安全的情況下共用型別，同時在執行"
-"緒間傳遞型別參考。"
+"這定義簡單的表示，若一個型別可以在確保執行緒安全的情況下被共用，這型別的參考值也可以安全的被傳遞於其他的執行緒。"
```
Comment (hueich) [inline po/zh-TW.po:14582]: May need to reformat.

File: po/zh-TW.po [hunk 12]
```diff
@@ -14988,9 +14986,9 @@ msgid ""
 msgstr ""
 "* `v` 已同時納入 `Arc` 和 `Mutex`，因為兩者的考量互不相關。\n"
 "* 將 `Mutex` 納入 `Arc`，是在執行緒間共享可變狀態的常見模式。\n"
-"* `v: Arc<_>` 需要複製成 `v2` 克隆，才能轉移到其他執行緒。請注意，`move` 已新"
+"* `v: Arc<_>` 需要複製成 `v2` ，才能轉移到其他執行緒。請注意，`move` 已新"
```
Comment (hueich) [inline po/zh-TW.po:14989]:
```suggestion
"* `v: Arc<_>` 需要複製成 `v2`，才能轉移到其他執行緒。請注意，`move` 已新"
```
Fixed spacing.

**Review by hueich — APPROVED** [review]
LGTM

**Comment by hueich** [comment]
Btw, why does the title say "morning"?  It's a nice greeting seeing this in the morning though. :)

**Comment by mgeisler** [comment]
> Btw, why does the title say "morning"? It's a nice greeting seeing this in the morning though. :)

😄  The day about concurrency has a morning part about classical concurrency and an afternoon part about async/await concurrency.

**Pull request merged**